### PR TITLE
Included more support for the new version of the WellIndexCalculation submodule

### PR DESCRIPTION
**Pull request by agchitu**
Hi Einar, sorry for this late request. I had the impression that these were already in.
Next week I'll spend more time on this and try to answer to your questions.

### CODE REVIEW

**Comment by einar90** [comment]
Hi Alin, it's no big problem, I haven't had much time to work on the code the last weeks anyway.

I tried to compile now, but I'm getting some compilation errors, both in the WIC unit tests and in FieldOpt Model library. It seems like all the errors are caused by some methods now taking vectors of cells instead of single cells as arguments. 

Could you update the unit tests for WIC to work with the new signatures, and check that the tests still pass? After that I can take care of updating the Model library to work with the new WIC; it should only be a batter of changing the `getWellBlock` method in `Model::Wells::Wellbore::WellSpline` to work with the new data structure.

File: FieldOpt/Reservoir/grid/cell.cpp [hunk 1]
```diff
@@ -52,24 +53,35 @@ namespace Reservoir {
...
+//
+
             int face_indices_points[6][4] = {
-                    {0, 2, 1, 3},
-                    {4, 5, 6, 7},
-                    {0, 4, 2, 6},
-                    {1, 3, 5, 7},
-                    {0, 1, 4, 5},
-                    {2, 6, 3, 7}
+                    {2, 0, 3, 1},
```
Comment (einar90) [inline FieldOpt/Reservoir/grid/cell.cpp]: Are you certain about these new indices? The reservoir unit tests are failing with these indices.

Comment (agchitu) [inline FieldOpt/Reservoir/grid/cell.cpp]: Hi Einar, 

(I'm not working on Mondays). Both indices are useful, I have new commits that will make sure the correct permutation is picked. It matters what type of grid the user is providing, namely if it right-handed of left-handed. For the unit tests the permutation that you have used is good but for the grids I had to work the one I included is working. It boils down to whether the z-axis is increasing toward the center of the Earth or towards the sky. In conclusion I will say to keep the permutation that is useful for the unit tests and I will commit in a new PR the code that makes sure the proper permutation is chosen by looking at the grid provided by the user. Cheers, Alin

Comment (einar90) [inline FieldOpt/Reservoir/grid/cell.cpp]: Aha, then I get it! I'll just keep the old definition for now then (with your commented out), just to keep the tests working for now.

Looks like I'll have both your pull requests merged by the end of today. How do you want me to do it? (ref: the email I sent you yesterday).

**Pull request merged**